testing_and_setup/compass/ocean/global_ocean/scripts/create_E3SM_coupling_files.py

```diff
@@ -1142,6 +1147,162 @@ def remap_rignot(inFileName, meshFileName, meshName, outFileName,
+            config.get('mesh', 'email')))
+
+    # determine date string
+    date_string = config.get('main', 'date_string')
+    if date_string == 'autodetect':
+        now = datetime.now()
+        date_string = now.strftime("%y%m%d")
+        print("- date string autodetected from today's date: {}".format(
+            date_string))
+        config.set('main', 'date_string', date_string)
+    else:
+        print("- date string specified in config file: {}".format(date_string))
+
+    # determine creation date
+    date_string = config.get('mesh', 'creation_date')
+    if date_string == 'autodetect':
+        now = datetime.now()
+        date_string = now.strftime("%m/%d/%Y %H:%M:%S")
+        print("- creation date autodetected from today's date: {}".format(
+            date_string))
+        config.set('mesh', 'creation_date', date_string)
+    else:
+        print("- creation date specified in config file: {}".format(
+            date_string))
+
+    # determine the maximum depth of the ocean
+    if config.get('mesh', 'max_depth') == 'autodetect':
+        with xr.open_dataset('mesh_before_metadata.nc') as dsMesh:
+            max_depth = dsMesh.refBottomDepth.isel(nVertLevels=-1).values
+            # round to the nearest 0.1 m
+            max_depth = np.round(max_depth, 1)
+
+        print("- maximum ocean depth autodetected mesh file: {}".format(
+            max_depth))
+        config.set('mesh', 'max_depth', '{}'.format(max_depth))
+    else:
+        max_depth = config.getfloat('mesh', 'max_depth')
+        print("- maximum ocean depth in config file: {}".format(max_depth))
+
+    # determine the number of vertical levels in the ocean
+    if config.get('mesh', 'levels') == 'autodetect':
+        with xr.open_dataset('mesh_before_metadata.nc') as dsMesh:
+            levels = dsMesh.sizes['nVertLevels']
+
+        print("- number of vertical levels in the ocean autodetected mesh file:"
+              " {}".format(levels))
+        config.set('mesh', 'levels', '{}'.format(levels))
+    else:
+        levels = config.getfloat('mesh', 'levels')
+        print("- number of vertical levels in the ocean in config file: "
+              "{}".format(levels))
+
+    # determine mesh name.  We do this last because other config options may
+    # be part of the mesh name
+    mesh_name = config.get('mesh', 'long_name')
+    currentDir = os.getcwd()
+    if mesh_name == 'autodetect':
+        path = currentDir.split('/')
+        if 'global_ocean' in path:
+            index = path.index('global_ocean') + 1
+            mesh_name = 'o{}'.format(path[index])
+            print("- mesh name autodetected from path: {}".format(mesh_name))
+            config.set('mesh', 'long_name', mesh_name)
+        else:
+            raise ValueError("mesh name not found in path. Please specify "
+                             "the mesh_name in config_E3SM_coupling_files.ini.")
+    else:
+        print("- mesh name specified in config file: ", mesh_name)
```

Comment:
would be nice to see both long and short name here.

Reply:
Okay, I can add this.

Reply:
Fixed.